### PR TITLE
added recurse_dicom_dir to heudiconv

### PR DESCRIPTION
**Pull request by tjhendrickson**
Added recurse_dicom_dir argument (--r, --recurse_dicom_dir) to run.py and logic to parser.py, in order so sub-directories are automatically recursed within the dicom_dir_template

### CODE REVIEW

File: heudiconv/cli/run.py [hunk 1]
```diff
@@ -137,6 +137,8 @@ def get_parser():
                        '(can be compressed) are supported in addition to '
                        'directory. All matching tarballs for a subject are '
                        'extracted and their content processed in a single pass')
+    group.add_argument('-r', '--recurse_dicom_dir', dest='recurse_dicom_dir',
+                       help='automatically recurse dicomdir sub directories to find all dicom files')
```
Comment (yarikoptic) [inline heudiconv/cli/run.py]: note that it is relevant only in the case of using a template... I think the better and somewhat easier solution is just to support '**' globbing and demand python 3.5 for that.  `recursive` kwarg to glob could be set if '**' is detected in the glob pattern

Comment (tjhendrickson) [reply heudiconv/cli/run.py]: Hmm. I was under the impression that heudiconv was under python 2.7 support, thus this is the reason I chose to do it the way I did. Wouldn't switching to python 3.5 but more burdensome  overall than using perhaps a sub-optimal way to recurse through the dicom directories? 

Am I missing something?

Comment (mgxd) [reply heudiconv/cli/run.py]: another option would be integrating [glob2](https://github.com/miracle2k/python-glob2) to support `**` globbing in py2/3

Comment (tjhendrickson) [reply heudiconv/cli/run.py]: I am in a bit of a time crunch for this. Could we merge the '--recurse_dicom_dir'?

Comment (mgxd) [reply heudiconv/cli/run.py]: @tjhendrickson tests are not passing yet, but perhaps we don't need to add the additional argument - we can just make the recursive behavior "smart" in that if the `dicom_dir_template` contains `**`, we recurse. If not, we go about with a normal `glob` call

Comment (tjhendrickson) [reply heudiconv/cli/run.py]: Oh I see. I can look into this tomorrow.

**Comment by tjhendrickson** [comment]
Fingers crossed that the checks pass!

**Comment by tjhendrickson** [comment]
My changes did not push for some reason...weird. Before I do I realize that I need to include glob2, is there a way or a file I should modify in order to install glob2?

**Comment by mgxd** [comment]
@tjhendrickson the `os.walk` implementation you previously integrated seems like a better solution (just to avoid adding additional dependencies) - I'd recommend writing a helper function that either uses `os.walk` or `glob` to fetch the matching files in `dicom_dir_template`

**Comment by mgxd** [comment]
@tjhendrickson are you still interested in this PR? You should be able to achieve this behavior with the `--files <directory>` flag, which will search all sub-directories of `<directory>`

**Comment by mgxd** [comment]
@tjhendrickson going to close this since it has gone stale - feel free to open if you wish to keep working on it/

**Pull request closed without merging**